### ISSUE TITLE
Ctrl-F mapping in mswin.vim

### ISSUE THREAD

**Issue by muddebutz** [vim/vim#1457]
Is there a reason why the find command `/` isn't mapped to `Ctrl+F` in the mswin.vim?

**Comment by zhen-huan-hu** [comment]
Would make more sense to map to promptfind then.

**Comment by chrisbra** [comment]
but `:promptfind` only makes sense in a gui version.

**Comment by chrisbra** [comment]
we then should probably also map `<c-h>` for `:%s` or `:promptrepl`

**Comment by zhen-huan-hu** [comment]
> but :promptfind only makes sense in a gui version.

Yeah, can use a condition to check has('gui_win32')

**Comment by chrisbra** [comment]
posted a patch [here](https://groups.google.com/d/msg/vim_dev/5nqjg4uJQLk/DWygtlrjDgAJ)

**Comment by zhen-huan-hu** [comment]
It should be C-H for replacing.

**Comment by zhen-huan-hu** [comment]
> posted a patch here

I would also suggest:

```
xnoremap <C-f> y:promptfind <C-r>"<CR>
xnoremap <C-h> y:promptrepl <C-r>"<CR>
```

**Comment by chrisbra** [comment]
fixed in [8.0.321](https://github.com/vim/vim/releases/tag/v8.0.0321)

**Comment by muddebutz** [comment]
Are the changes in every new release or only in 8.0.321?

**Comment by muddebutz** [comment]
I installed your release, but if I stroke `ctrl+f` in insert mode nothing happened and if I stroke `ctrl+f` in normal mode it displays `:promptfind` in the command line but I'm running vim not gvim.

**Comment by muddebutz** [comment]
I installed your release, but if I stroke `ctrl+f` in insert mode nothing happened and if I stroke `ctrl+f` in normal mode it displays `:promptfind` in the command line but I'm running vim not gvim.

**Comment by plbowers** [comment]
Do not, not, not like this change. CTRL-F is for page down - without it I have to take my hands off the home-row keys to find the other pgdn key. This goes against the whole philosophy of vim, IMHO.

I like to be able to use ctrl-c/x/v for copy/paste and some of the other mswin behaviors, but this one is no good. Especially since you've got a LARGE user base using this it's just too big a change.

Set it back and allow for some kind of OPTIONAL configuration that would do this. Don't make everybody that likes a little mswin compatibility swallow the loss of page-down!

**Comment by plbowers** [comment]
What is the preferred way of getting rid of this? Do I have to go in and edit my mswin.vim file directly? Isn't that something you don't want people to do so as to keep updates easy? Suggestions how I can dump these 2 new keymaps?

**Comment by chrisbra** [comment]
you can threw in a VimEnter autocommand, that will remove those mappings. Something  like this: `:au VimEnter * exe "unmap <c-f>"|exe "iunmap <c-f>"| exe "cunmap <c-f>"`

**Comment by plbowers** [comment]
I couldn't get the au to work, but I just put unmap/iunmap/cunmap on separate lines in my Prog.../Vim/_vimrc after the mswin.vim and it solved the problem. Thanks for your help!

I see this as a workaround - I really think that redefining ctrl-F by default for all Windows vim users is not a good idea.

**Comment by kyledmoore** [comment]
I just want to express my displeasure with this change.  I've been using the Windows version of gVim for nearly 20 years now and use vim in no small part due to it allowing me to keep my hands on the home keys during file navigation.  Remapping ctrl-f like this was quite jarring when I installed a new version on a VM I'm using for work, only slightly less worse then perhaps changing the 'hjkl' keys.  I think this should have been relegated to the easy mode if done at all.

I get that you are dealing with competing requests and priorities, but this one goes too far toward the realm of amalgamation with lesser editors.  I see it as a departure from a core vi philosophy to start mucking with standard navigation keys as part of OS-specific mappings.

I'm simply adding an `unmap<C-F>` line to my _vimrc following `behave mswin`, to fix, so this might come across as whiny, but just want to speak up about it while simultaneously thanking all for the hard work on the product and humbly requesting that these types of subjective decisions are made w/ a broader philosophical viewpoint about what this editor represents in mind.  I think if you ask yourself that question, the decision to skip this change would have been obvious as one can fire up any vim editor on a variety of OS and use ctrl-f, ctrl-b to jump pages as a core navigational technique.  It's nothing short of vi canon, except now on Windows.

Someone mentioned the defaulting of the clipboard commands as an argument for this change, but I don't think that's a great argument considering that the clipboard shortcuts are truly an OS-specific peculiarity vs. one particular to a given editor as this change is.  Furthermore, I would argue that a newbie coming to gVim, trying to search with ctrl-F and finding that doesn't work is going to end up searching to find the proper way (/) and that might serve as a gateway for them to learn how to use the proper features of the editor, which is necessary to develop an appreciation of it.  Those that would dismiss it up front over this are likely to abandon it for a more conventional editor, regardless.

**Comment by tonymec** [comment]
``behave mswin`` by itself only changes the values of four options, it doesn't define any mappings. Maybe you meant ``runtime mswin.vim`` instead? That script does define mappings, several of which hide good "classical Vim" key bindings, and for my part I avoid that script like the plague, but, well, to each his own.

About the ``:behave`` statement, I set the associated settings to something which is neither ``behave mswin`` nor ``behave xterm`` but somewhere in between, as follows:
```
if exists('+selectmode')
  set slm=mouse,key
endif
if exists('+mousemodel')
  set mousemodel=popup
endif
if exists('+keymodel')
  set keymodel=startsel
endif
if exists('+selection')
  set selection=inclusive
endif
```
Use it or not: one of the things I like best in Vim is its customizability — Vim can be made to do what you like, not only what its developer likes.

Best regards,
Tony.

**Comment by chrisbra** [comment]
@kyledmoore note, this has been fixed as of c3fdf7f80b2febdd8a8f7a
